sync/class.jetpack-sync-sender.php

```diff
@@ -61,7 +61,16 @@ public function do_sync() {
 		}
 
 		// don't sync if we are throttled
-		if ( $this->get_next_sync_time() > microtime( true ) ) {
+		$next_sync_queue = $this->get_next_sync_time();
+		if ( $next_sync_queue > microtime( true ) ) {
+			// Even though we shouldn't be syncing untill next time if the queue is really small
+			// lets try any way.
+			if ( $this->sync_queue->size() < 12 && $next_sync_queue < $this->get_sync_wait_time() + time() ) {
```

Comment:
I am not sure we need to do this: `$next_sync_queue < $this->get_sync_wait_time() + time()`